source/sigv4.c

```diff
@@ -1033,3 +1820,239 @@ SigV4Status_t SigV4_AwsIotDateToIso8601( const char * pDate,
 
     return returnStatus;
 }
+
+SigV4Status_t SigV4_GenerateHTTPAuthorization( const SigV4Parameters_t * pParams,
+                                               char * pAuthBuf,
+                                               size_t * authBufLen,
+                                               char ** pSignature,
+                                               size_t * signatureLen )
+{
+    SigV4Status_t returnStatus = SigV4Success;
+    CanonicalContext_t canonicalContext;
+    char * pPath = NULL, *pAlgorithm = NULL;
+    size_t pathLen = 0U, encodedLen = 0U, algorithmLen = 0U, authPrefixLen = 0U;
+    HmacContext_t hmacContext = { 0 };
+    SigV4String_t signingKey;
+
+    /*returnStatus = verifySigV4Parameters( pParams ); */
+
+    /* Default arguments. */
+    if( ( pParams->pAlgorithm == NULL ) || ( pParams->algorithmLen == 0 ) )
```

Comment:
`pParams` wasn't checked for NULL-ness. Is that what `verifySigV4Params` will be for?